pan3d/catalogs/__init__.py

```diff
@@ -0,0 +1,14 @@
+import importlib
+
+
+def call_catalog_function(catalog_name, function_name, **kwargs):
+    try:
+        module = importlib.import_module(f"pan3d.catalogs.{catalog_name}")
+        func = getattr(module, function_name)
+        return func(**kwargs)
+    except ImportError:
+        raise ValueError(
+            f"{catalog_name} catalog module not enabled. Install pan3d[{catalog_name}] to load this catalog."
+        )
+    except AttributeError:
+        raise ValueError(f"{catalog_name} is not a valid catalog module.")
```

Comment:
why don't you add a `load_dataset`, `get_catalog_search_options` and `get_catalog` in that same file and just import those in the dataset builder. That way they will look like regular methods. 

You can also use __all__ to make that `call_catalog_function` method internal.